src/main/java/md2k/mCerebrum/cStress/features/ECGFeatures.java

```diff
@@ -582,13 +578,11 @@ public DataPoint rr_ave_update(List<DataPoint> rpeak_temp1, DataPointStream rr_a
                 for (int i = peak_interval.size() - AUTOSENSE.PEAK_INTERVAL_MINIMUM_SIZE; i < peak_interval.size(); i++) {
                     result.value += peak_interval.get(i);
                 }
-                result.value /= 8.0;
-                result.timestamp = rpeak_temp1.get(rpeak_temp1.size() - 1).timestamp;
-                rr_ave.add(result);
+                rr_ave.add(new DataPoint(rpeak_temp1.get(rpeak_temp1.size() - 1).timestamp, result.value /= 8.0));
```

Comment:
This is weird logic.  Why are you updating `result.value` insides of a function call to create a new datapoint?